src/core/qgsvectorfilewriter.cpp

```diff
@@ -2442,15 +2442,56 @@ class QgsVectorFileWriterMetadataContainer
       datasetOptions.clear();
       layerOptions.clear();
 
+      QStringList compressionMethods;
+      const QStringList searchCompressions =
+      {
+        QStringLiteral( "NONE" ),
+        QStringLiteral( "SNAPPY" ),
+        QStringLiteral( "BROTLI" ),
+        QStringLiteral( "ZSTD" )
+      };
+
+      if ( GDALDriverH hParquetDrv = GDALGetDriverByName( "PARQUET" ) )
+      {
+        if ( const char *xml = GDALGetMetadataItem( hParquetDrv, GDAL_DS_LAYER_CREATIONOPTIONLIST, nullptr ) )
+        {
+          for ( const QString &comp : searchCompressions )
+          {
+            QRegularExpression re( "<Value[^>]*>\\s*" + comp + "\\s*</Value>" );
+            if ( re.match( QString::fromUtf8( xml ) ).hasMatch() )
+            {
+              compressionMethods << comp;
+            }
+          }
+        }
+      }
+
       layerOptions.insert( QStringLiteral( "COMPRESSION" ), new QgsVectorFileWriter::SetOption(
                              QObject::tr( "Compression method." ),
-                             QStringList()
-                             << QStringLiteral( "UNCOMPRESSED" )
-                             << QStringLiteral( "SNAPPY" ),
+                             compressionMethods,
                              QStringLiteral( "SNAPPY" ), // Default value
                              false // Allow None
                            ) );
 
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,12,0)
+      layerOptions.insert( QStringLiteral( "COMPRESSION_LEVEL" ), new QgsVectorFileWriter::IntOption(
+                             QObject::tr( "Compression level." ),
+                             0 // Default value
```

Comment:
There was no default value previously, but I didn't realize that QGIS required one. Just fixed in GDAL per https://github.com/OSGeo/gdal/commit/97063060ed05411e0b531527eda1a76dbee0e69c so that -1 means default

```suggestion
                             -1 // Default value
```

Reply:
0 is a valid value for Brotli ... IMHO we can afford "stealing" -1 from ZSTD, as using negative values is really a very niche use case

Reply:
just realized `gdal --format PARQUET` shows BROTLI actually has level `0` :+1:

so I just add `-1` as default value to sync with GDAL :smile: